tests/unit/contracts/test_standalone.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+import pytest
+from grimp.adaptors.graph import ImportGraph
+
+from importlinter.application.app_config import settings
+from importlinter.contracts.standalone import StandaloneContract
+from importlinter.domain.contract import ContractCheck
+from tests.adapters.printing import FakePrinter
+from tests.adapters.timing import FakeTimer
+
+
+@pytest.fixture(scope="module", autouse=True)
+def configure():
+    settings.configure(TIMER=FakeTimer())
+
+
+class TestStandaloneContract:
+    def _build_default_graph(self):
+        graph = ImportGraph()
+        for module in (
+            "mypackage",
+            "mypackage.blue",
+            "mypackage.blue.alpha",
+            "mypackage.blue.beta",
+            "mypackage.blue.beta.foo",
+            "mypackage.blue.foo",
+            "mypackage.blue.hello",
+            "mypackage.blue.world",
+            "mypackage.green",
+            "mypackage.green.bar",
+            "mypackage.yellow",
+            "mypackage.yellow.gamma",
+            "mypackage.yellow.delta",
+            "mypackage.other",
+            "mypackage.other.sub",
+            "mypackage.other.sub2",
+        ):
+            graph.add_module(module)
+        return graph
+
+    def _check_default_contract(self, graph):
```

Comment:
Not a blocker, but IMO it would be better if we passed in the set of modules that we want to be standalone from the test itself - they're a bit hidden in here.